File: MainGame/game/inGame/map/scripts/Utils/BackEndUtil.cs
using Godot;

namespace ZeromaXPlayground.game.inGame.map.scripts.Utils;

public static class BackEndUtil
{
    public static System.Numerics.Vector2 To(Vector2I vec) => new(vec.X, vec.Y);

    public static Vector2I From(System.Numerics.Vector2 vec) => new((int)vec.X, (int)vec.Y);
}

File: MainGame/FrontEnd4IdleStrategyFS/Common/BackEndUtil.fs
namespace FrontEnd4IdleStrategyFS.Common

module BackEndUtil =
    open Godot

    let toBackEnd (vec: Vector2I) = (vec.X, vec.Y)

    let fromBackEndI (x: int, y: int) = Vector2I(x, y)

    let fromBackEnd (x: int, y: int) = Vector2(float32 x, float32 y)